### PR TITLE
Update srp getting started guide and code sample for track 2 SDK

### PR DESCRIPTION
**Pull request by pauljewellmsft**
In addition to code sample and getting started guide changes, I upgraded the code sample project to .NET 6 in order to use certain C# language features.

### CODE REVIEW

File: AzureStorageNew.csproj [hunk 1]
```diff
@@ -1,121 +1,32 @@
...
-    <Compile Include="Properties\AssemblyInfo.cs" />
+    <Reference Include="System.IdentityModel" />
+    <Reference Include="System.Net" />
+    <Reference Include="System.Security" />
   </ItemGroup>
   <ItemGroup>
-    <None Include="App.config" />
-    <None Include="packages.config" />
+    <PackageReference Include="Azure.Identity" Version="1.6.0" />
+    <PackageReference Include="Azure.ResourceManager.Storage" Version="1.0.0-beta.11" />
```
Comment (blueww) [inline AzureStorageNew.csproj:20]: This should be changed to the GA version when the SDK GA.

File: README.md [hunk 2]
```diff
@@ -38,67 +38,84 @@ To run the sample, follow these steps:
...
+using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Azure;
+using Azure.Identity;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Storage;
+using Azure.ResourceManager.Storage.Models;
+using Azure.Core;
```
Comment (blueww) [inline README.md:58]: Should the using be sort in the doc?

File: README.md [hunk 3]
```diff
@@ -107,30 +124,37 @@ The sample lists all of the storage accounts in a given subscription:
...
 
 ```csharp
 //Get the storage account keys for a given account and resource group
-IList<StorageAccountKey> acctKeys = storageMgmtClient.StorageAccounts.ListKeys(rgName, accountName).Keys;
+StorageAccountGetKeysResult result = storageAccount.GetKeys();
+IReadOnlyList<StorageAccountKey> acctKeys = result.Keys;
+Console.WriteLine($"List of storage account keys in {storageAccount.Id.Name}:");
+foreach (StorageAccountKey acctKey in acctKeys)
+{
+    Console.WriteLine($"\t{acctKey.KeyName}");
```
Comment (blueww) [inline README.md:148]: Should we print only the key name but not key vault here? (normally, customer will care about key value more)
As the before code doesn't show how to print the keys, we might also don't need to print it?

File: README.md [hunk 4]
```diff
@@ -139,30 +163,19 @@ The storage account SKU specifies what type of replication applies to the storag
...
-};
-var storageAccount = storageMgmtClient.StorageAccounts.Update(rgname, acctName, parameters);
+StorageSku updateSku = new StorageSku(StorageSkuName.StandardLrs);
+StorageAccountCreateOrUpdateContent updateParams = new StorageAccountCreateOrUpdateContent(updateSku, kind, location);
+await accountCollection.CreateOrUpdateAsync(WaitUntil.Completed, storAccountName, updateParams);
 ```
 
 Note that modifying the SKU for a production storage account may have associated costs. For example, if you convert a locally redundant storage account to a geo-redundant storage account, you will be charged for replicating your data to the secondary region. Before you modify the SKU for a production account, be sure to consider any cost implications. See [Azure Storage replication](https://azure.microsoft.com/documentation/articles/storage-redundancy/) for additional information about storage replication.
 
-### Check storage account name availability
```
Comment (blueww) [inline README.md:151]: Why no Track2 sample for this?
We might can add a tracks SDK sample for this, the API should looks like :
```
subscription.CheckStorageAccountNameAvailability(new StorageAccountNameAvailabilityContent("accountname"));
```

**Comment by pauljewellmsft** [comment]
Thanks for the review @blueww. Comments have been addressed - note that once the SDK is GA I'll come back and update the package before I publish these changes.

**Review by ArcturusZhang — COMMENTED** [review]
Looks like we are mixing sync method and async method. Should we unify them to all use the async version or all use the sync version?

File: Program.cs [hunk 5]
```diff
@@ -40,191 +41,127 @@ public class StorageAccountTests
...
-        private static void CreateStorageAccount(string rgname, string acctName, StorageManagementClient storageMgmtClient)
-        {                                                                       
-            StorageAccountCreateParameters parameters = GetDefaultStorageAccountParameters();
-
-            Console.WriteLine("Creating a storage account...");
-            var storageAccount = storageMgmtClient.StorageAccounts.Create(rgname, acctName, parameters);
-            Console.WriteLine("Storage account created with name " + storageAccount.Name);                                                                     
+            AsyncPageable<StorageAccountResource> storAccts = resourceGroup.GetStorageAccounts().GetAllAsync();
+            Console.WriteLine($"List of storage accounts in {resourceGroup.Id.Name}:");
+            await foreach (StorageAccountResource storAcct in storAccts)
```
[the remark this answers is not in the document]
Comment (pauljewellmsft) [reply Program.cs]: Thanks for the suggestion @ArcturusZhang. I've updated the sample.

**Pull request merged**